### PR TITLE
Upgrade calling sdk for Teams interop branch to 2.0.0-beta.1

### PR DESCRIPTION
**Pull request by jimchou-dev**
## Purpose

* Upgrade calling sdk for Teams interop branch to 2.0.0-beta.1
* Refactor breaking change for recording and transcription notification

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ x ] Other... Please describe: SDK upgrade
```

## What to Check
Verify that the following are valid
* Verify after the upgrade the teams meeting recording/transcription notification still shows

## Other Information

* The new calling SDK removes the dependency for AzureCore, need to add it in Podfile as work around as suggested by https://github.com/Azure/Communication/issues/344 have a separate ticket in ADO 2533262 for tracking

### CODE REVIEW

File: AzureCalling/External/Calling/CallingContext.swift [hunk 1]
```diff
@@ -129,6 +131,11 @@ class CallingContext: NSObject {
                 self.callType = joinConfig.callType
                 self.displayName = joinConfig.displayName
                 self.isCameraPreferredOn = joinConfig.isCameraOn
+
+                self.callRecordingFeature = self.call?.api(RecordingFeature.self)
+                self.callTranscriptionFeature = self.call?.api(TranscriptionFeature.self)
```
Comment (AmyL219) [inline AzureCalling/External/Calling/CallingContext.swift:136]: Do you have the link to the change log?

Comment (jimchou-dev) [reply AzureCalling/External/Calling/CallingContext.swift:136]: https://github.com/Azure/Communication/releases/

**Review by AmyL219 — APPROVED** [review]

**Comment by palatter** [comment]
Looks good, testing now.

**Comment by palatter** [comment]
> Looks good, testing now.

All good!

**Review by palatter — APPROVED** [review]

**Pull request merged**